File: src/client/src/data/models/transaction.ts
export default class Transaction{
    constructor(public id:number, public user_id:number, public txcode: string,
        public amount:number, public status:string, public purchase_id: number){
            this.id = id;
            this.user_id = user_id;
            this.txcode = txcode;
            this.amount = amount;
            this.status  = status;
            this.purchase_id = purchase_id;
    }
}

File: src/server/src/legarda/api/models/transaction.py
from legarda.api import db, flask_bcrypt
from sqlalchemy.sql import func


class Transaction(db.Model):
    """ Transaction model """

    __tablename__ = "transactions"

    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
    status = db.Column(db.String(10), nullable=False, default='init')
    reference = db.Column(db.String(100), nullable=False)
    amount = db.Column(db.Float, nullable=False, default=0)
    date_created = db.Column(db.DateTime, nullable=False, default=func.now())
    purchase_id = db.Column(db.Integer, db.ForeignKey('purchases.id'),nullable=False)
    

    def __init__(self, reference, status='init'):
        self.reference = reference
        self.status = status

    def __repr__(self):
        return "<Transaction %s %s>"%(self.reference, self.status.upper())
